### PR TITLE
enhancement-DataGridComponent

### PR DESCRIPTION
**Pull request by karolkolodziej**
Added DataGrid Component.

To do:

- [x] TS
- [x] Tests
   - [x] Constructor
   - [x] Events
- [x] Fix demo
- [x] Sync handler

### CODE REVIEW

File: ts/Extensions/DashboardPlugin/DataGridComponent.ts [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+ */
+class DataGridComponent extends Component<DataGridComponent.ChartComponentEvents> {
+
+    /* *
+     *
+     *  Static properties
+     *
+     * */
+
+    public static syncHandlers = HighchartsSyncHandlers;
```
Comment (goransle) [inline ts/Extensions/DashboardPlugin/DataGridComponent.ts:56]: Question here might be if we should build out these existing handlers to support the datagrid, or have a separate file for the DataGridSyncHandlers 🤔

**Comment by goransle** [comment]
@karolkolodziej looks like a good setup. Don't know if I have too much to add to the todo list.

File: ts/Extensions/DashboardPlugin/DataGridComponent.ts (hunk 1 above)
Comment (karolkolodziej) [reply ts/Extensions/DashboardPlugin/DataGridComponent.ts:56]: Synch handler is not yet implemented but that is actually a good question! And I think it depends on what we would like to synchronize. But I guess it should be a separate file as you suggested. @bre1470 @pawellysy

Comment (pawellysy) [reply ts/Extensions/DashboardPlugin/DataGridComponent.ts:56]: Separate files sound better IMO

File: ts/Dashboard/Component/SharedComponentState.ts [hunk 2]
```diff
@@ -230,13 +232,20 @@ Serializable<SharedComponentState, SharedComponentState.JSON> {
     }
 
     public setHoverPoint(
-        point?: SharedComponentState.PresentationHoverPointType,
-        eventDetail?: {}
+        point?: SharedComponentState.PresentationHoverPointType | HTMLElement,
+        eventDetail?: any
```
Comment (bre1470) [inline ts/Dashboard/Component/SharedComponentState.ts:236]: `any` should be fixed, as an object with `isDataGrid` property is used. At least it should become `AnyRecord`.

**Review by bre1470 — CHANGES_REQUESTED** [review]

**Comment by bre1470** [comment]
Otherwise it looks good.

**Comment by bre1470** [comment]
Thank you! Looks good. 👍

**Review by bre1470 — APPROVED** [review]

**Review by goransle — APPROVED** [review]
Nice!

**Review by pawellysy — APPROVED** [review]
Looks good to me :D

**Pull request merged**